include/swift/Basic/SupplementaryOutputPaths.h

```diff
@@ -146,6 +146,12 @@ struct SupplementaryOutputPaths {
   /// The output path to generate ABI baseline.
   std::string ABIDescriptorOutputPath;
 
+  /// The output path for YAML optimization record file.
```

Comment:
I've preferred to keep track of both because there are separate file types registered, `yaml-opt-record` and `bitstream-opt-record` and that allows the possibility for the supplemental outputs file to have an entry for both.

Also I would consider it weird if we treat the file types interchangeable, for example if you specify `bitstream` format in the flag it should then use the right file type entry from the supplemental outputs list, not pick whatever is there (e.g. `bitstream` is specified but is written out to the `yaml` file type.

It'd probably be simpler if there was only one file type registered, ignoring format, from the start, like `opt-record`, but not sure if it's ok to make such a change.

Reply:
I agree, the file types make that a bit more complicated.

This is fine as is then, thanks!